main/src/ca/uwaterloo/flix/language/errors/TypeError.scala

```diff
@@ -107,6 +162,7 @@ object TypeError {
          |>> Unable to unify the Boolean formulas: '${red(FormatType.formatType(baseType1))}' and '${red(FormatType.formatType(baseType2))}'.
```

Comment:
Much like my previous comment on `GeneralizationError`, I think "Unable to unify Boolean formulas" is a very cryptic message. Maybe we could simply write "Mismatched purity" in the summary and use same phrasing in the message or at least let the user know that this is the case in the `explain` function.
Same applies to `MismatchedTypes` and `OccursCheckError`

Reply:
Yes. I will deal with these later.